File: sprint2/C++/lesson11code2.cpp
#include <iostream>

int factorial(int n) {
    int accumulator = 1;
    int i = n;
    while (i > 1) {
        accumulator *= i;
        i -= 1;
    }
    return accumulator;
}


int main() {
    std::cout << factorial(5) << std::endl;  // 120
    return 0;
}

File: sprint2/By languages/go/lesson11code2.go
package main

import "fmt"

func factorial(n int) int {
    accumulator := 1
    i := n
    for i > 1 {
        accumulator *= i
        i -= 1
    }
    return accumulator
}

func main() {
    fmt.Println(factorial(5))   // 120
}
